web-ui/src/pages/ReceivedRequestsPage.jsx

```diff
@@ -98,55 +114,92 @@ const ReceivedRequestsPage = () => {
           type: UPDATE_TOAST,
           payload: {
             severity: "error",
-            toast: res.error
+            toast: res.error,
           },
         });
       }
-    }
+    };
 
     if (csrf && currentUserId) {
       getAllFeedbackRequests().then((data) => {
         if (data) {
-          setReceivedRequests(data.filter((req) => req.submitDate === undefined));
-          setSubmittedRequests(data.filter((req) => req.submitDate && req.submitDate.length === 3));
-          setIsLoading(false)
+          setCanceledRequests(data.filter((req) => req.status === "canceled"));
+          setReceivedRequests(
+            data.filter(
+              (req) => req.submitDate === undefined && req.status !== "canceled"
+            )
+          );
+          setSubmittedRequests(
+            data.filter(
+              (req) =>
+                req.submitDate &&
+                req.submitDate.length === 3 &&
+                req.status !== "canceled"
+            )
+          );
+          setIsLoading(false);
         }
       });
     }
-
   }, [csrf, currentUserId]);
-
   useEffect(() => {
+    let filteredCanceled = [...canceledRequests];
     let filteredReceived = [...receivedRequests];
     let filteredSubmitted = [...submittedRequests];
 
     // Search for intersection of multiple queries separated by commas
-    const queries = searchText.split(",").map((search) => search.trim().toLowerCase());
+    const queries = searchText
+      .split(",")
+      .map((search) => search.trim().toLowerCase());
 
     if (searchText.trim()) {
       for (let query of queries) {
-        filteredReceived = filteredReceived.filter((request) => {
-          const creatorName = selectProfile(state, request.creatorId).name.toLowerCase();
-          const requesteeName = selectProfile(state, request.requesteeId).name.toLowerCase();
-          return creatorName.includes(query) || requesteeName.includes(query);
-        });
+        const setFiltered = (filteredOption) => {
+          filteredOption = filteredOption.filter((request) => {
+            const creatorName = selectProfile(
+              state,
+              request.creatorId
+            ).name.toLowerCase();
+            const requesteeName = selectProfile(
+              state,
+              request.requesteeId
+            ).name.toLowerCase();
+            return creatorName.includes(query) || requesteeName.includes(query);
+          });
+        };
+        setFiltered(filteredCanceled);
+        setFiltered(filteredReceived);
+        setFiltered(filteredSubmitted);
+        // filteredCanceled = filteredCanceled.filter((request) => {
+        //   const creatorName = selectProfile(state, request.creatorId).name.toLowerCase();
+        //   const requesteeName = selectProfile(state, request.requesteeId).name.toLowerCase();
+        //   return creatorName.includes(query) || requesteeName.includes(query);
+        // });
```

Comment:
It looks like the goal of this code was to set `filteredCanceled`, `filteredReceived`, and `filteredSubmitted` to their filtered versions based on the query string. However, it seems that `filteredOption` is only a local parameter within the lambda method. As a result, setting `filteredOption = filteredOption.filter(...)` only modifies it locally. To achieve the desired effect, you could try returning `filteredOption` from the `setFiltered` function and calling it like this instead: `filteredCanceled = setFiltered(filteredCanceled);` This should allow you to modify `filteredCanceled`, `filteredReceived`, and `filteredSubmitted` as desired.